sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -836,6 +836,15 @@ export class JsonRpcProvider {
     );
   }
 
+  // NOTE: This does not use the `sui_getChainIdentifier` method as it's not yet widely available.
+  async getChainIdentifier(): Promise<string> {
+    const checkpoint = await this.getCheckpoint({ id: '0' });
+    const bytes = fromB58(checkpoint.digest);
+    return [...bytes.slice(0, 4)]
+      .map((byte) => byte.toString(16).padStart(2, '0'))
```

Comment:
why do we need padStart?

Reply:
Because `0`.toString(16) is `0`, but I think we want `00` in that case. But I'm reworking these to use our existing hex formatting anyway so this will go away